File: picture-server/src/main/java/com/applory/pictureserver/domain/request/RequestController.java
package com.applory.pictureserver.domain.request;

import com.applory.pictureserver.shared.Result;
import lombok.RequiredArgsConstructor;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

import javax.validation.Valid;


@RestController
@RequestMapping("/api/v1/requests")
@RequiredArgsConstructor
public class RequestController {

    private final RequestService requestService;

    @PostMapping("")
    @ResponseStatus(HttpStatus.CREATED)
    public void createRequest(@Valid @RequestBody RequestDto.Create dto) {
        requestService.createRequest(dto);
    }

    @GetMapping("")
    public Page<RequestDto.VM> getRequests(RequestDto.Search search, Pageable pageable) {
        return requestService.getRequests(search, pageable);
    }

    @GetMapping("/{id}")
    public Result getRequest(@PathVariable String id) {
        return Result.success(requestService.getRequest(id));
    }
}

File: picture-server-kt/src/main/kotlin/com/applory/pictureserverkt/request/RequestController.kt
package com.applory.pictureserverkt.request

import org.springframework.data.domain.Page
import org.springframework.data.domain.Pageable
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*
import java.util.*
import javax.validation.Valid

@RestController()
@RequestMapping("/api/v1/requests")
class RequestController(private val requestService: RequestService) {

    @PostMapping("")
    @ResponseStatus(HttpStatus.CREATED)
    fun createRequest(@Valid @RequestBody dto: RequestDto.Create): RequestDto.VM {
        return RequestDto.convertToVM(requestService.createRequest(dto))
    }

    @GetMapping("")
    fun getRequests(search: RequestDto.Search, pageable: Pageable): Page<RequestDto.VM> {
        return requestService.getRequests(search, pageable).map { RequestDto.convertToVM(it) }
    }

    @GetMapping("/{id}")
    fun getRequests(@PathVariable id: UUID): RequestDto.VM {
        return requestService.getRequest(id)
    }

}
